inlang/packages/paraglide/paraglide-js/src/compiler/runtime/extract-locale-from-request.test.ts

```diff
@@ -286,7 +288,55 @@ test("returns locale from custom strategy", async () => {
 		},
 	});
 
-	const locale = runtime.extractLocaleFromRequest(request);
+	const locale = await runtime.extractLocaleFromRequest(request);
+	expect(locale).toBe("fr");
+});
+
+test("returns locale from custom strategy with an asynchronous function", async () => {
+	const runtime = await createParaglide({
+		blob: await newProject({
+			settings: {
+				baseLocale: "en",
+				locales: ["en", "fr", "de"],
+			},
+		}),
+		strategy: ["custom-header", "baseLocale"],
+	});
+
+	class FakeDB {
+		db = new Map<string, string>();
+		constructor() {
+			this.db.set("1", "fr");
+		}
+
+		async getUserLocaleById(id: string) {
+			setTimeout(() => null, 4);
```

Comment:
```suggestion
			setTimeout(() => null, 4);
```

This could be removed. If you declare a method to be `async`, it's enough for it to be `async`.